app/controllers/application_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ApplicationController < ActionController::Base
+    helper_method :current_user_session, :current_user
+    helper_method :display_post_time
+
+    private
+
+    def require_user
+      unless current_user
+        flash[:notice] = "You must be logged in to access this page"
+        redirect_to new_user_session_path
+        return false
+      end
+    end
+
+    def require_user_admin
+      unless current_user.admin
+        flash[:notice] = "You must be logged in as admin to access this page"
+        redirect_to new_user_session_path
+        return false
+      end
+    end
+
+    def display_post_time(time_at_post)
```

Comment:
This is great! I ended up using a gem for relative time but it is quite unreliable...